lib/rpc/server/peer.go

```diff
@@ -238,15 +256,17 @@ func (r serverPeer) Reconnect(ctx context.Context) (Client, error) {
 
 	_, err = clt.PeerJoin(ctx, &pb.PeerJoinRequest{Addr: r.addr, Config: &r.config, SystemInfo: payload})
 	if err != nil {
-		return nil, &backoff.PermanentError{trace.Wrap(err)}
```

Comment:
I found this to be the wrong decision that the peer's Reconnect was deciding whether to abort the reconnects - instead I moved this to the default `ReconnectStrategy` implementation.